lib/ansible/modules/windows/win_psmodule.ps1

```diff
@@ -90,97 +348,123 @@ Function Remove-Repository{
     }
 }
 
-Function Install-PsModule {
-    param(
-      [Parameter(Mandatory=$true)]
-      [string]$Name,
-      [string]$Repository,
-      [bool]$AllowClobber,
-      [bool]$CheckMode
-    )
-    if (Get-Module -Listavailable|?{$_.name -eq $Name}){
-        $result.output = "Module $($Name) already present"
-    }
-    else {      
-      try{
-        # Install NuGet Provider if needed
-        Install-NugetProvider -CheckMode $CheckMode;
-
-        $ht = @{
-            Name      = $Name;
-            WhatIf    = $CheckMode;
-            ErrorAction = "Stop";
-            Force     = $true;
-        };
-
-        # If specified, use repository name to select module source
-        if ($Repository) {
-            $ht["Repository"] = "$Repository";
-        }
-
-        # Check Powershell Version (-AllowClobber was introduced in PowerShellGet 1.6.0)
-        if ("AllowClobber" -in ((Get-Command PowerShellGet\Install-Module | Select -ExpandProperty Parameters).Keys)) {
-          $ht['AllowClobber'] = $AllowClobber;
-        }
-        
-        Install-Module @ht | out-null;
-        
-        $result.output = "Module $($Name) installed"
-        $result.changed = $true
-      }
-      catch{
-        $ErrorMessage = "Problems installing $($Name) module: $($_.Exception.Message)"
+# Check PowerShell version, fail if < 5.0 and required modules are not installed
+$PsVersion = $PSVersionTable.PSVersion
+if ($PsVersion.Major -lt 5 ) {
+    $PrereqModulesInstalled = Install-PrereqModule -TestInstallationOnly
+    if ( -not $PrereqModulesInstalled ) {
+        $ErrorMessage = "Modules PowerShellGet and PackageManagement in versions 1.6.0 and 1.1.7 respectively have to be installed before using the win_psmodule."
         Fail-Json $result $ErrorMessage
-      }
     }
 }
 
-Function Remove-PsModule {
-    param(
-      [Parameter(Mandatory=$true)]
-      [string]$Name,
-      [bool]$CheckMode
-    )
-    # If module is present, unistalls it.
-    if (Get-Module -Listavailable|?{$_.name -eq $Name}){
-      try{
-        Uninstall-Module -Name $Name -Confirm:$false -Force -ErrorAction Stop -WhatIf:$CheckMode | out-null
-        $result.output = "Module $($Name) removed"
-        $result.changed = $true
-      }
-      catch{
-        $ErrorMessage = "Problems removing $($Name) module: $($_.Exception.Message)"
+if ( $required_version -and ( $minimum_version -or $maximum_version ) ) {
```

Comment:
You don't have to move across to it yet but you might want to consider looking at `Ansible.Basic`, it is able to handle scenarios like mutually exclusive options and required_if scenarios just like Python's basic.py.

Reply:
I would not like to rewrite that module now again.